ensime_shared/ensime.py

```diff
@@ -136,16 +136,23 @@ def queue_poll(self, sleep_t=0.5):
 
         Value of sleep is low to improve responsiveness.
         """
+        connection_alive = True
         while self.running:
             if self.ws:
                 def logger_and_close(m):
                     self.log("Websocket exception: {}".format(m))
-                    self.teardown()
+                    if not self.running:
```

Comment:
Show me some suggestions and I'll be pleased to add them to the PR. Specifically, what do you mean with being more selective?‎

I'm thinking that maybe a good idea would be not to close and exit vim if such a `WebsocketException` happens and, instead, we just disable `ensime-vim` (via runtimepath) and notify the user. This is probably the cleanest solution.